geomesa-filter/src/main/scala/org/locationtech/geomesa/filter/FilterHelper.scala

```diff
@@ -209,9 +209,10 @@ object FilterHelper extends LazyLogging {
 
       // Note: although not technically required, all known spatial predicates are also binary spatial operators
       case f: BinarySpatialOperator if isSpatialFilter(f) =>
+        if (attribute == null) logger.warn("Geometry attribute is not set.  Will include all geometries from filter")
```

Comment:
That was at Anthony's suggestion. If there are two differently named geometry attributes in the filter, then both geometries get returned which might not be what the caller intends. He figured it might be prudent to log a warning.